### PR TITLE
form-task

### PR DESCRIPTION
**Pull request by OlgaFrontend**
### [Demo](https://olgafrontend.github.io/kottans-frontend/form_task/dist/) |  [Code](https://github.com/OlgaFrontend/kottans-frontend/tree/master/form_task)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/OlgaDubynka/form_task/src/index.html [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+    <meta http-equiv="Content-type" content="text/html; charset=utf-8"/>
+    <meta http-equiv="X-UA-Compatible" content="ie=edge" />
+    <meta name="viewport"
+          content="width=device-width, user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0" />
+    <title>form</title>
+    <link rel="stylesheet" href="css/style.css">
+</head>
+<body>
+  <div class="root">
+      <header></header>
```
Comment (A-Ostrovnyy) [inline submissions/OlgaDubynka/form_task/src/index.html]: `<header>` content where are you?

File: submissions/OlgaDubynka/form_task/src/index.html [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+          <a class="sign-in" href="#">
+              <span class="sign-in-icon"></span>
+              <span class="sign-in-text">sign in</span>
+          </a>
+        </div>
+        <div class="form-body">
+          <section class="form-intro">
+            <div class="your-order">
+              <div class="order-box">
+                <p class="order-name">your order</p>
```
Comment (A-Ostrovnyy) [inline submissions/OlgaDubynka/form_task/src/index.html]: Maybe better use headings `h1-h6` for this text.

File: submissions/OlgaDubynka/form_task/src/index.html [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+                </ul>
+              </div>
+              <div class="processing">
+                <button class="proc-btn proc-btn_green proc-btn_mb" type="submit">place order</button>
+                <button class="proc-btn" type="button">print a quote</button>
+              </div>
+            </div>
+            <div class="support">
+              <p class="support-title">help center</p>
+              <div class="support-desc">If you need assistance while placing your order, contact one of our beloved Project Manager by phone or email.</div>
```
Comment (A-Ostrovnyy) [inline submissions/OlgaDubynka/form_task/src/index.html]: Better wrap text by `<p>` tag.

File: submissions/OlgaDubynka/form_task/src/index.html [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+              <address class="support-contacts">
+                <a class="support-link" href="tel:+3800555555">+380 (0) 555 555</a>
+                <a class="support-link" href="mailto:help@qtest.com">help@qtest.com</a>
+              </address>
+            </div>
+          </section>
+          <section class="options">
+            <div class="container">
+              <h1 class="title-h1">select what you want to translate</h1>
+              <div class="group">
```
Comment (A-Ostrovnyy) [inline submissions/OlgaDubynka/form_task/src/index.html]: These radio-buttons is required. Also, one of them must be checked by default.

File: submissions/OlgaDubynka/form_task/src/index.html [hunk 5]
```diff
@@ -0,0 +1,300 @@
...
+              </ul>
+            </div>
+          </section>
+          <section class="lang">
+            <div class="container">
+              <h3 class="title-h3"><span class="title-number">2</span>Choose your languages pairings</h3>
+              <div class="choose-from">
+                <span class="lang-label">from</span>
+                <div class="select-wrap">
+                  <select name="choose-lang-from" id="from-lang" class="choose-lang">
```
Comment (A-Ostrovnyy) [inline submissions/OlgaDubynka/form_task/src/index.html]: This input is required.

File: submissions/OlgaDubynka/form_task/src/index.html [hunk 6]
```diff
@@ -0,0 +1,300 @@
...
+                      <option value="Arabic">Arabic</option>
+                      <option value="Ukrainian">Ukrainian</option>
+                      <option value="Kurdish">Kurdish</option>
+                  </select>
+                </div>
+              </div>
+              <div class="choose-from">
+                <span class="lang-label">to</span>
+                <div class="select-wrap">
+                    <select name="choose-lang-to" id="to-lang-1" class="choose-lang">
```
Comment (A-Ostrovnyy) [inline submissions/OlgaDubynka/form_task/src/index.html]: This input is required too.

File: submissions/OlgaDubynka/form_task/src/index.html [hunk 7]
```diff
@@ -0,0 +1,300 @@
...
+                    </select>
+                </div>          
+                <button class="btn-del" type="button"></button>
+                <button class="btn-add" type="button">&plus;  add</button>
+              </div>
+            </div>
+          </section>
+          <div class="wrapper-section">
+            <div class="container">
+              <section class="tone">
```
Comment (A-Ostrovnyy) [inline submissions/OlgaDubynka/form_task/src/index.html]: Radio-buttons inside this section is required. Also, one of them must be checked by default.

File: submissions/OlgaDubynka/form_task/src/index.html [hunk 8]
```diff
@@ -0,0 +1,300 @@
...
+                <input type="radio" id="media" name="tone"  class="group-input visually-hidden">
+                <label for="media" class="group-label">
+                    <span class="group-border"></span>
+                    <span class="group-text group-media">Media / Formal</span>
+                </label>
+              </section>
+              <section class="urgent">
+                <h3 class="title-h3"><span class="title-number">4</span>Tell us how urgent this work is</h3>
+                <label for="urgent-range" class="range-label">
+                  <input type="range" name="urgent-range" id="urgent-range" class="range-input">
```
Comment (A-Ostrovnyy) [inline submissions/OlgaDubynka/form_task/src/index.html]: This input is required. Also, add dots on the `runnable-track`.

File: submissions/OlgaDubynka/form_task/src/index.html [hunk 9]
```diff
@@ -0,0 +1,300 @@
...
+                  <a href="#" class="sign-in-link">Sign in.</a>
+              </p>
+              <div class="user-info">
+                  <label for="fullname" class="sign-up-label">
+                    <span class="sign-up-text">fullname</span>
+                    <input type="text" id="fullname" class="sign-up-input" placeholder="Mathieu Mayer" title="please enter your fullname" required>
+                  </label>
+                  <label for="email" class="sign-up-label">
+                    <span class="sign-up-text">email address</span>
+                    <input type="email" id="email" class="sign-up-input" placeholder="mathieu.mm@me.com"
```
Comment (A-Ostrovnyy) [inline submissions/OlgaDubynka/form_task/src/index.html]: Check validation of the email input. [Image](https://imgur.com/a/tboEzHW) - this is valid value.

File: submissions/OlgaDubynka/form_task/src/index.html [hunk 10]
```diff
@@ -0,0 +1,300 @@
...
+            </div>
+          </div>
+          <section class="payment">
+            <div class="container">
+              <h3 class="title-h3"><span class="title-number">5</span>Sign up and choose a payment method</h3>
+              <p class="sign-in-text">
+                  Already have an account with us?
+                  <a href="#" class="sign-in-link">Sign in.</a>
+              </p>
+              <div class="user-info">
```
Comment (A-Ostrovnyy) [inline submissions/OlgaDubynka/form_task/src/index.html]: Add styles on `:valid` and `:invalid` state.

File: submissions/OlgaDubynka/form_task/src/index.html [hunk 11]
```diff
@@ -0,0 +1,300 @@
...
+                    <span class="sign-up-text">email address</span>
+                    <input type="email" id="email" class="sign-up-input" placeholder="mathieu.mm@me.com"
+                    title="please enter your email example@gmail.com" required>
+                  </label>
+                  <label for="password" class="sign-up-label">
+                    <span class="sign-up-text">Choose a password</span>
+                    <input type="password" name="password" id="password" class="sign-up-input" pattern="^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?!.*\s).*$" placeholder="Must be (A-Z, a-z and 0-9)" required title="password must be (A-Z, a-z and 0-9)">
+                  </label>
+              </div>
+              <div class="cards-wrap">
```
Comment (A-Ostrovnyy) [inline submissions/OlgaDubynka/form_task/src/index.html]: Radio-buttons inside this section is required. Also, one of them must be checked by default.

File: submissions/OlgaDubynka/form_task/src/index.html [hunk 12]
```diff
@@ -0,0 +1,300 @@
...
+                  <input type="radio" id="amazon" name="card" class="group-input visually-hidden">
+                  <label for="amazon" class="group-label">
+                      <span class="group-border"></span>
+                      <img class="group-img" src="images/amazon.png" alt="amazon">
+                  </label>
+              </div>
+              <div class="sign-up-card">
+                  <label for="card-number" class="sign-up-label">
+                    <span class="sign-up-text">Card number</span>
+                    <input class="sign-up-input  card-number" type="text" name="card-number" id="card-number" pattern="[0-9]{13,16}" placeholder="Must be 16-18 digits"
```
Comment (A-Ostrovnyy) [inline submissions/OlgaDubynka/form_task/src/index.html]: Add `maxlength ` attribute.

File: submissions/OlgaDubynka/form_task/src/index.html [hunk 13]
```diff
@@ -0,0 +1,300 @@
...
+              what we do
+            </h4>
+            <p class="about-us-text">
+              We provide a full range of language services including content development, document, website and social media eccounts translations.
+            </p>
+          </div>
+          <div class="location">
+            <h4 class="footer-title">our main office</h4>
+            <address class="location-address">
+              <span>Dubfi, UAE</span><br>
```
Comment (A-Ostrovnyy) [inline submissions/OlgaDubynka/form_task/src/index.html]: Avoid using `<br>` for similar markup.

File: submissions/OlgaDubynka/form_task/src/scss/_form.scss [hunk 14]
```diff
@@ -0,0 +1,1363 @@
...
+      margin-bottom: 20px;
+    }
+  }
+}
+
+.go-back,
+.sign-in {
+  padding: 0 40px;
+  display: flex;
+  display: flex;
```
Comment (A-Ostrovnyy) [inline submissions/OlgaDubynka/form_task/src/scss/_form.scss]: Duplicating properties.

File: submissions/OlgaDubynka/form_task/src/scss/_form.scss [hunk 15]
```diff
@@ -0,0 +1,1363 @@
...
+  .sign-in-icon {
+    width: 17px;
+    height: 17px;
+    margin-right: 15px;
+    background-repeat: no-repeat;
+    background-size: 100%;
+  }
+
+  .arrow-icon {
+    background-image: url('/images/arrow-back.png');
```
Comment (A-Ostrovnyy) [inline submissions/OlgaDubynka/form_task/src/scss/_form.scss]: Images are missing.

File: submissions/OlgaDubynka/form_task/src/index.html (hunk 1 above)
Comment (OlgaFrontend) [reply submissions/OlgaDubynka/form_task/src/index.html]: Thanks I removed it)

File: submissions/OlgaDubynka/form_task/src/index.html (hunk 2 above)
Comment (OlgaFrontend) [reply submissions/OlgaDubynka/form_task/src/index.html]: yes I've done it

File: submissions/OlgaDubynka/form_task/src/index.html (hunk 3 above)
Comment (OlgaFrontend) [reply submissions/OlgaDubynka/form_task/src/index.html]: yes thanks)

File: submissions/OlgaDubynka/form_task/src/scss/_form.scss (hunk 14 above)
Comment (OlgaFrontend) [reply submissions/OlgaDubynka/form_task/src/scss/_form.scss]: thanks) done

File: submissions/OlgaDubynka/form_task/src/index.html (hunk 4 above)
Comment (OlgaFrontend) [reply submissions/OlgaDubynka/form_task/src/index.html]: thanks) done

File: submissions/OlgaDubynka/form_task/src/index.html (hunk 5 above)
Comment (OlgaFrontend) [reply submissions/OlgaDubynka/form_task/src/index.html]: done

File: submissions/OlgaDubynka/form_task/src/index.html (hunk 6 above)
Comment (OlgaFrontend) [reply submissions/OlgaDubynka/form_task/src/index.html]: thanks) done

File: submissions/OlgaDubynka/form_task/src/index.html (hunk 7 above)
Comment (OlgaFrontend) [reply submissions/OlgaDubynka/form_task/src/index.html]: thanks) I fixed it

File: submissions/OlgaDubynka/form_task/src/index.html (hunk 8 above)
Comment (OlgaFrontend) [reply submissions/OlgaDubynka/form_task/src/index.html]: I added dots)

File: submissions/OlgaDubynka/form_task/src/index.html (hunk 9 above)
Comment (OlgaFrontend) [reply submissions/OlgaDubynka/form_task/src/index.html]: done

File: submissions/OlgaDubynka/form_task/src/index.html (hunk 11 above)
Comment (OlgaFrontend) [reply submissions/OlgaDubynka/form_task/src/index.html]: done

File: submissions/OlgaDubynka/form_task/src/scss/_form.scss (hunk 14 above)
Comment (OlgaFrontend) [reply submissions/OlgaDubynka/form_task/src/scss/_form.scss]: done

File: submissions/OlgaDubynka/form_task/src/scss/_form.scss (hunk 15 above)
Comment (OlgaFrontend) [reply submissions/OlgaDubynka/form_task/src/scss/_form.scss]: done

File: submissions/OlgaDubynka/form_task/src/index.html (hunk 13 above)
Comment (OlgaFrontend) [reply submissions/OlgaDubynka/form_task/src/index.html]: done

File: submissions/OlgaDubynka/form_task/src/index.html (hunk 12 above)
Comment (OlgaFrontend) [reply submissions/OlgaDubynka/form_task/src/index.html]: done

File: submissions/OlgaDubynka/form_task/src/index.html (hunk 10 above)
Comment (OlgaFrontend) [reply submissions/OlgaDubynka/form_task/src/index.html]: done

File: submissions/OlgaDubynka/form_task/src/index.html [hunk 16]
```diff
@@ -0,0 +1,300 @@
...
+<body>
+  <div class="root">
+      <header></header>
+      <form action="#" method="POST" class="form">
+        <div class="form-head">
+          <a class="go-back" href="#">
+              <span class="arrow-icon"></span>
+              <span class="go-back-text">go back</span>
+          </a>
+          <a class="logo"><img src="images/logo.png" alt="logo-of-company"></a>
```
Comment (A-Ostrovnyy) [inline submissions/OlgaDubynka/form_task/src/index.html]: Your logo is not accessible from the keyboard. Add attribute `href='/'`.

File: submissions/OlgaDubynka/form_task/src/scss/_form.scss [hunk 17]
```diff
@@ -873,10 +872,31 @@
 }
 
 .range-input {
+  position: relative;
   display: block;
   width: 413px;
 }
 
+.range-input:before,
```
Comment (A-Ostrovnyy) [inline submissions/OlgaDubynka/form_task/src/scss/_form.scss]: Height of dots must be same as runnable track

File: submissions/OlgaDubynka/form_task/src/index.html [hunk 18]
```diff
@@ -0,0 +1,300 @@
...
+              <span>Dubfi, UAE</span><br>
+              <span>P.O.Box: 502735</span><br>
+              <a href="tel:+3800555555" class="phone">+380 (0) 555 555</a>
+            </address>
+          </div>
+          <div class="socials">
+            <h4 class="footer-title">
+              connect with us
+            </h4>
+            <a href="#" class="fb">Facebook</a>
```
Comment (A-Ostrovnyy) [inline submissions/OlgaDubynka/form_task/src/index.html]: Social icons are missing.

Comment (OlgaFrontend) [reply submissions/OlgaDubynka/form_task/src/index.html]: done) thanks

File: submissions/OlgaDubynka/form_task/src/index.html (hunk 16 above)
Comment (OlgaFrontend) [reply submissions/OlgaDubynka/form_task/src/index.html]: done) thanks

File: submissions/OlgaDubynka/form_task/src/scss/_form.scss (hunk 17 above)
Comment (OlgaFrontend) [reply submissions/OlgaDubynka/form_task/src/scss/_form.scss]: done) thanks

**Comment by A-Ostrovnyy** [comment]
Well done!

**Pull request merged**